packages/components/Badge/src/useBadge.ts

```diff
@@ -1,12 +1,15 @@
-import type { BadgeProps } from './Badge.types';
+import type { BadgeInfo, BadgeProps } from './Badge.types';
 
-export const useBadge = (props: BadgeProps): BadgeProps => {
-  if (!props) return {};
+export const useBadge = (props: BadgeProps): BadgeInfo => {
+  if (!props) return { props: {}, state: {} };
```

Comment:
Why would this happen?

Reply:
Before the return type just contained "BadgeProps". To handle the state variables as well (required for the toggle selection), have created a new type "BadgeInfo" consisting of "BadgeProps" and "BadgeState"